### PR TITLE
[MM-35625] Add controller for changing cloud subscription

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
This endpoint is needed for the front-end to call, so that users can change cloud plan they're subscribed to. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-35625

#### Release Note
```release-note
None
```

### CODE REVIEW

**Review by AGMETEOR — APPROVED** [review]

File: api4/cloud.go [hunk 1]
```diff
@@ -70,6 +71,47 @@ func getSubscription(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PERMISSION_SYSCONSOLE_WRITE_BILLING) {
+		c.SetPermissionError(model.PERMISSION_SYSCONSOLE_WRITE_BILLING)
+		return
+	}
+
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		c.Err = model.NewAppError("Api4.updateCloudCustomer", "api.cloud.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (ethervoid) [inline api4/cloud.go:87]:
```suggestion
		c.Err = model.NewAppError("Api4.changeSubscription", "api.cloud.app_error", nil, err.Error(), http.StatusBadEquest)
```

File: api4/cloud.go [hunk 2]
```diff
@@ -70,6 +71,47 @@ func getSubscription(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	var subscriptionChange *model.SubscriptionChange
+	if err = json.Unmarshal(bodyBytes, &subscriptionChange); err != nil {
+		c.Err = model.NewAppError("Api4.changeSubscription", "api.cloud.app_error", nil, err.Error(), http.StatusBadRequest)
+	}
+
+	currentSubscription, appErr := c.App.Cloud().GetSubscription("")
```
Comment (ethervoid) [inline api4/cloud.go:96]:
```suggestion
	currentSubscription, appErr := c.App.Cloud().GetSubscription(c.AppContext.Session().UserId)
```

**Review by ethervoid — APPROVED** [review]
Some small changes

**Pull request merged**